models/issue.go

```diff
@@ -1731,6 +1731,13 @@ func SearchIssueIDsByKeyword(kw string, repoID int64, limit, start int) (int64,
 }
 
 func updateIssue(e Engine, issue *Issue) error {
+	if issue.PosterID <= 0 {
+		log.Error("On issue %d gitea try to set %d as PosterID", issue.ID, issue.PosterID)
```

Comment:
I still think it would have been better to do the log after we tried reloading the posterID.

That way you could have two log messages:

1. Gitea has a bug and is doing a stupid thing trying to update a posterID to -1 but we were able to stop it from doing harm.
2. There's an issue out there with a posterID -1 and so your data is corrupt and you need to fix it.